PayItForward.ConsoleClient.UnitTests/Web/Controllers/StoriesController/Donate_Should.cs

```diff
@@ -0,0 +1,251 @@
+                    CollectedAmount = 900,
+                    Title = "help"
+                }, this.storyId);
+
+            // Assert
+            this.usersService.Verify(m => m.GetUserById("username"), Times.Once);
+        }
+
+        [Fact]
+        public void ReturnViewResult_WithDonateViewModel()
+        {
+            // Arrange
+            this.usersService.Setup(story => story.GetUserById("username"))
+                .Returns(this.HelperUserDTO());
+
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns(this.HelperStoryDto().FirstOrDefault());
+
+            this.storiesController.ControllerContext = new ControllerContext
+            {
+                HttpContext = new DefaultHttpContext
+                {
+                    User = new ClaimsPrincipal(
+                    new ClaimsIdentity(
+                        new Claim[]
+                        {
+                            new Claim(ClaimTypes.NameIdentifier, "username")
+                        },
+                        "someAuthTypeName"))
+                }
+            };
+            var donationDTO = new DonationDTO()
+            {
+                Amount = 10,
+                Donator = new UserDTO(),
+                Story = this.HelperStoryDto().FirstOrDefault()
+            };
+
+            this.donationService.Setup(d => d.Add(donationDTO, this.storyId)).Returns(true);
+
+            // Act
+            var result = this.storiesController.Donate(
+                new DonateViewModel()
+                {
+                    Amount = 10,
+                    Donator = new UserDTO(),
+                    CollectedAmount = 900,
+                    Title = "help"
+                }, this.storyId);
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.IsType<DonateViewModel>(viewResult.ViewData.Model);
+        }
+
+        private List<StoryDTO> HelperStoryDto()
+        {
+            return new List<StoryDTO>()
+            {
+                new StoryDTO
+                {
+                    Category = new CategoryDTO()
+                    {
+                        Name = "Health"
+                    },
+                    CollectedAmount = 3,
+                    CreatedOn = DateTime.UtcNow,
+                    Description = "Some description",
+                    Title = "Title",
+                    User = this.HelperUserDTO(),
+                    Id = this.storyId
+                }
+            };
+        }
+
+        private UserDTO HelperUserDTO()
+        {
+            return new Models.UserDTO()
+            {
+                Email = "vicky.penkova@gmial.com",
+                FirstName = "Viki",
+                LastName = "Penkova",
+                AvilableMoneyAmount = 100,
+                Id = "alabala"
+            };
+        }
+
+        private DonateViewModel HelperDonateViewModel()
```

Comment:
HelperDonateViewModel can be renamed to GetDonateViewModel because it is method and it should talk about what it is